flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/BoundedMultiInput.java

```diff
@@ -19,13 +19,25 @@
 
 import org.apache.flink.annotation.PublicEvolving;
 
-/** Interface for the multi-input operators that can process EndOfInput event. */
+/**
+ * Interface for multi-input operators that need to be notified about the logical/semantical end of
+ * input.
+ *
+ * <p><b>WARNING:</b> It is not safe to use this method to commit any transactions or other side
+ * effects! You can use this method to e.g. flush data buffered for the given input or implement an
+ * ordered reading from multiple inputs via {@link InputSelectable}.
```

Comment:
Wouldn't it be better to put this paragraph before `endInput` method?

Reply:
Makes sense.